CSC8503/DebugNetworkedGame.cpp

```diff
@@ -307,16 +344,18 @@ void DebugNetworkedGame::SpawnPlayers(){
             mServerPlayers.emplace(i, nullptr);
         }
     }
+    int playerPeerId = 0;
     if (mThisServer) {
-        mLocalPlayer = mServerPlayers[0];
+        mLocalPlayer = mServerPlayers[playerPeerId];
     }
     else
     {
-        mLocalPlayer = mServerPlayers[GetPlayerPeerID()];
+        playerPeerId = GetPlayerPeerID();
+        mLocalPlayer = mServerPlayers[playerPeerId];
     }
+      mServerPlayers[playerPeerId]->SetIsLocalPlayer(true);
     tempPlayer = (PlayerObject*)mLocalPlayer;
-    //localPlayer->GetRenderObject()->SetVisibility(false);
-    LockCameraToObject(mLocalPlayer);
+    mLocalPlayer->SetActive();
```

Comment:
This might merit some thought. Currently, setting something to 'inactive' means it just won't get rendered, but usually in systems, 'inactive' also means they won't undergo any physics calculations either. This can be useful functionality for stuff we want to destroy'(collectibles, the heist item, some effect a buff provides), not least because we can also easily make them active again. We get the performance benefit of not doing any calculations, and we don't have to deal with removing an already instantiated object.

If we go down this route with the local player being inactive to not render them, this locks us in to that pattern. Perhaps some other mechanism would work?

Reply:
Yes we should implement this in the near future. We can create a story for it and discuss it in the sprint meeting 👍 
And we should add a parameter to the SetActive function 😆 🤣